pages/2_Data.py

```diff
@@ -20,22 +20,44 @@
 def change_photo_state():
     st.session_state["photo"]="done"
 
-uploaded_file = col2.file_uploader("Choose a CSV file", on_change=change_photo_state)
-file_images = []
-url_column_name = "change to the name of the column with url listing"
+uploaded_file = col2.file_uploader("Choose a CSV file", on_change=None)
+
+dataset_names = []
+dataset_locations = []
+
+name_column_name = "Dataset_Name"
+url_column_name = "Location"
 
 if uploaded_file:
     df = pd.read_csv(uploaded_file)
-    url_list = df[url_column_name]
-    for url in url_list:
-        file_images.append(url)
+    name_list = df[name_column_name]
+    location_list = df[url_column_name]
+    for name in name_list:
+        dataset_names.append(name)
+    for location in location_list:
+        dataset_locations.append(location)
+
+    with st.expander("Click to choose datasets"):
+        for name in dataset_names:
+            j = 0
+            st.write(name)
+            show = st.checkbox("choose dataset", key=name)
+            if show:
+                for image in os.listdir(dataset_locations[j]):
```

Comment:
logic could be simpler with `enumerate` here?

Reply:
I changed the key into j, but the key is not used here. The key is a necessary argument to create a checkbox. The variable "name" is the name of the dataset, which in this case is Human_Detection.